### ISSUE TITLE
So2sat Data Module

### ISSUE THREAD

**Issue by AbdelghaniKABOT** [microsoft/torchgeo#1249]
### Summary

I would like to request features for the [So2Sat data module](https://github.com/microsoft/torchgeo/blob/main/torchgeo/datamodules/so2sat.py). 

- Currently, it does not have the option to load only the RGB bands, so we can fine-tuning models. Could you please add this functionality?
- Additionally, it would be helpful to have the option for binary labels of the RGB bands (Building from 0 to 9 and Non-Building from 10 to 16). This would enable binary classification or semantic segmentation using the Lightning trainer.

**Comment by adamjstewart** [comment]
The first one wouldn't be hard to implement. Would you like to submit a PR?

The second one feels more specific. It might be better to create your own subclass of the datamodule to add this functionality.

**Comment by AbdelghaniKABOT** [comment]
@adamjstewart  
What do you think about this 
```
"""Binary Classes of  RGB So2Sat datamodule."""

from typing import Any
import torch

from torchgeo.datasets import So2Sat
from torchgeo.trainers import NonGeoDataModule


class So2SatRGB(So2Sat):
    def __getitem__(self, index: int) -> Any:
        x = super().__getitem__(index)
        # Select only the RGB bands
        x = x[[8, 9, 10], :, :]
        # Convert labels to binary
        y = self.labels[index]
        y = 1 if y >= 1 and y <= 9 else 0
        return x, y


class So2SatDataModule(NonGeoDataModule):
    def __init__(
        self,
        batch_size: int = 64,
        num_workers: int = 0,
        **kwargs: Any,
    ) -> None:
        """Initialize a new So2SatDataModule instance.
        Args:
            batch_size: Size of each mini-batch.
            num_workers: Number of workers for parallel data loading.
            **kwargs: Additional keyword arguments passed to
                :class:`~torchgeo.datasets.So2Sat`.
        """
        super().__init__(So2SatRGB, batch_size, num_workers, **kwargs)

    def setup(self, stage: str) -> None:
        """Set up datasets.
        Called at the beginning of fit, validate, test, or predict. During distributed
        training, this method is called from every process across all the nodes. Setting
        state here is recommended.
        Args:
            stage: Either 'fit', 'validate', 'test', or 'predict'.
        """
        if stage in ["fit"]:
            self.train_dataset = So2SatRGB(split="train", **self.kwargs)
        if stage in ["fit", "validate"]:
            self.val_dataset = So2SatRGB(split="validation", **self.kwargs)
        if stage in ["test"]:
            self.test_dataset = So2SatRGB(split="test", **self.kwargs)
```

**Comment by adamjstewart** [comment]
I don't think we need a new datamodule, just one additional `band_set` option.

**Comment by AbdelghaniKABOT** [comment]
I also added a subclass to handle binary labels

**Comment by calebrob6** [comment]
@AbdelghaniKABOT 

You can get binary labels for your experiment with the following:

```
def binarize_labels(sample):
    if sample["label"] >= 1:
        sample["label"] = 1
    else:
        sample["label"] = 0
    return sample

dm = So2SatDataModule(
    root="path/to/your/data/",
    transforms=binarize_labels
)
```

**Comment by adamjstewart** [comment]
RGB option was added in #1283, binary labels can be created using transforms. I think this issue is now complete.
